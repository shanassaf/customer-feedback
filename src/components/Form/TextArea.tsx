import React from 'react';
import styled from 'styled-components';

interface Props {
	setFeedback: (feedback: Feedback) => void;
	feedback: Feedback;
}
const StyledTextArea = styled.textarea`
	padding: 10px 4px;
	cursor: pointer;

	:hover {
		border-color: #1b047c;
	}
`;

export const TextArea = ({ setFeedback, feedback }: Props) => {
	return (
		<StyledTextArea
			required
			name="comment"
			value={feedback.comment}
			onChange={(ev) => {
				setFeedback({ ...feedback, comment: ev.target.value });
			}}
		/>
	);
};
