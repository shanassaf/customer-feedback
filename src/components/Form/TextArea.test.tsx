import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { TextArea } from './TextArea';

describe('TextArea', () => {
	const feedback = {
		name: '',
		email: '',
		comment: '',
		rating: '0',
	} as Feedback;
	const props = {
		setFeedback: jest.fn(),
		feedback,
	};
	test('component renders correctly', () => {
		const { asFragment } = render(<TextArea {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
	test('component calls setFeedback on change', () => {
		const { getByRole } = render(<TextArea {...props} />);
		const comment = getByRole('textbox');
		fireEvent.change(comment, { target: { value: 'This is amazing' } });
		expect(props.setFeedback).toHaveBeenCalledWith({
			...feedback,
			comment: 'This is amazing',
		});
	});
});
