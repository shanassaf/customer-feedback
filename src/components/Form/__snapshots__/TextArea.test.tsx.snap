// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component renders correctly 1`] = `
<DocumentFragment>
  <textarea
    class="sc-bdnylx cDOOWf"
    name="comment"
    required=""
  />
</DocumentFragment>
`;
